src/material-experimental/mdc-form-field/_mdc-text-field-textarea-overrides.scss

```diff
@@ -10,7 +10,11 @@
 @mixin private-text-field-textarea-overrides() {
   // Ensures that textarea elements inside of the form-field have proper vertical spacing
   // to account for the floating label. Also ensures that there is no vertical text overflow.
-  .mat-mdc-textarea-input {
+  .mat-mdc-form-field-input-control.mat-mdc-textarea-input {
+    // Set the vertical alignment for textareas inside form fields to be the middle. This
+    // ensures that textareas do not stretch the infix container vertically without having
+    // multiple rows of text. See: https://github.com/angular/components/pull/22089.
+    vertical-align: middle;
```

Comment:
Are you sure this doesn't mess up prefix/suffix icon alignment?

Reply:
I've tested with a text suffix/prefix and icons, looks good. Here is a screenshot for you to verify:

Reply:
Is it just me or are there some extra lines in that screenshot? 
![122447133-b815fe00-cfa3-11eb-8865-0c7dd87bd938](https://user-images.githubusercontent.com/4450522/122450782-b3ebdf80-cfa7-11eb-8a26-922742fbfb63.png)

Reply:
This is unrelated to this PR (can be observed in Miles' form-field demo too; https://mmalerba-demo-1.web.app/mdc-input). 

I'll have a look at this too; it's also observable on the MDC textfield itself but not as obvious.

Reply:
Yeah I've noticed that extra line before too, it comes from the MDC styles. I remember having a similar issue back when I was writing our outline form field too.